codelists/definition.py

```diff
@@ -29,62 +34,104 @@ def __hash__(self):
 
     @classmethod
     def from_fragment(cls, fragment):
+        """Build DefinitionRule from string fragment.
+
+        The fragment will have one of the following formats:
+
+        * "XYZ" -- XYZ is included
+        * "XYZ<" -- XYZ is included with all its descendants
+        * "~XYZ" -- XYZ is excluded
+        * "~XYZ<" -- XYZ is excluded with all its descendants
+        """
+
         if fragment[0] == "~":
-            negated = True
+            code_is_excluded = True
             fragment = fragment[1:]
         else:
-            negated = False
+            code_is_excluded = False
 
         if fragment[-1] == "<":
-            includes_children = True
+            applies_to_descendants = True
             fragment = fragment[:-1]
         else:
-            includes_children = False
+            applies_to_descendants = False
 
-        return cls(fragment, negated=negated, includes_children=includes_children)
+        return cls(
+            fragment,
+            code_is_excluded=code_is_excluded,
+            applies_to_descendants=applies_to_descendants,
+        )
 
 
 class Definition:
-    def __init__(self, elements):
-        self.elements = set(elements)
+    """Represents a set of rules that define a list of codes.  Each rule indicates
+    whether a given code is included (an "including rule") or excluded (an "excluding
+    rule") in a list of codes, and whether the inclusion/exclusion applies to the code's
+    descendants too.
+    """
 
-    def negated_elements(self):
-        for e in self.elements:
-            if e.negated:
-                yield e
+    def __init__(self, rules):
+        self.rules = set(rules)
 
-    def unnegated_elements(self):
-        for e in self.elements:
-            if not e.negated:
-                yield e
+    def excluding_rules(self):
+        """Yield rules that exclude a code (and possible that code's descendants).
+        """
```

Comment:
Single line docstrings have the quotes on the same line.

Reply:
Tell that to the linter!